plan/initialize.go

```diff
@@ -275,13 +277,24 @@ func (p PhysicalTableReader) init(allocator *idAllocator, ctx context.Context) *
 	p.basePlan = newBasePlan(TypeTableReader, allocator, ctx, &p)
 	p.basePhysicalPlan = newBasePhysicalPlan(p.basePlan)
 	p.TablePlans = flattenPushDownPlan(p.tablePlan)
+	p.schema = p.tablePlan.Schema()
 	return &p
 }
 
 func (p PhysicalIndexReader) init(allocator *idAllocator, ctx context.Context) *PhysicalIndexReader {
 	p.basePlan = newBasePlan(TypeIndexReader, allocator, ctx, &p)
 	p.basePhysicalPlan = newBasePhysicalPlan(p.basePlan)
 	p.IndexPlans = flattenPushDownPlan(p.indexPlan)
+	if _, ok := p.indexPlan.(*PhysicalAggregation); ok {
+		p.schema = p.indexPlan.Schema()
+	} else {
+		schemaInKV := p.indexPlan.Schema()
+		is := p.IndexPlans[0].(*PhysicalIndexScan)
```

Comment:
I think I just confused `indexPlan` with `IndexPlans`, can we make the two names more different?
How about rename `IndexPlans` to `flattenedPlans`?